src/libpriv/rpmostree-core.c

```diff
@@ -3798,6 +3798,44 @@ rpmostree_context_get_kernel_changed (RpmOstreeContext *self)
   return self->kernel_changed;
 }
 
+/* We have a messy dance in dealing with /usr/etc and /etc; the
+ * current model is basically to have it be /etc whenever we're running
+ * any code.
+ */
+gboolean
+rpmostree_core_prepare_etc (int              rootfs_dfd,
+                            gboolean        *renamed_etc,
+                            GError         **error)
+{
+  if (!glnx_fstatat_allow_noent (rootfs_dfd, "usr/etc", NULL, 0, error))
+    return FALSE;
+  *renamed_etc = (errno == 0);
```

Comment:
Minor: we could avoid modifying the outvar upfront here without allocating another var with something like:

```c
if (errno == ENOENT)
  *renamed_etc = FALSE;
else
  {
    ...
    *renamed_etc = TRUE;
  }
return TRUE;
```

?

Reply:
Does modifying the outvar upfront seem bad here though?  I think it really only matters when allocating, right?

Reply:
I usually try not to modify outvars until we know we're returning `TRUE`. I find it helps mentally to reason about the code, even if you don't want to rely on that behaviour. Not a big deal of course, we do this in a bunch of places already.